schema.development.kf

```diff
@@ -653,10 +653,16 @@ action create_credentials_by_dwg(
         THEN ERROR('dwg_owner not found')
     END;
 
-    $ag_timelock = idos.parseRfc3339($dwg_access_grant_timelock); // Will fail if not in the RFC3339 format
+    $ag_timelock = idos.parse_rfc3339($dwg_access_grant_timelock); // Will fail if not in the RFC3339 format
     $times_validation = idos.validate_not_usable_times($dwg_not_before, $dwg_not_after); // Check the format and precedence
     SELECT CASE WHEN $times_validation != 1 THEN ERROR('dwg_not_before must be before dwg_not_after') END;
 
+    // Check it current block timestamp in time range allowed by write grant
+    SELECT CASE WHEN parse_unix_timestamp($dwg_not_before, 'YYYY-MM-DD"T"HH24:MI:SS"Z"')::int > @block_timestamp
+        OR parse_unix_timestamp($dwg_not_after, 'YYYY-MM-DD"T"HH24:MI:SS"Z"')::int < @block_timestamp
+        THEN ERROR('this write grant can be used after dwg_not_before and before dwg_not_after')
```

Comment:
Let's:
- factor out the affirmative part of the condition and then negate it; this makes the success case obvious.
- only use less than for ordering dates (so it appears written like the flow of time)
- clarify the error a little bit

```suggestion
    SELECT CASE WHEN NOT (
            parse_unix_timestamp($dwg_not_before, 'YYYY-MM-DD"T"HH24:MI:SS"Z"')::int < @block_timestamp
            OR @block_timestamp < parse_unix_timestamp($dwg_not_after, 'YYYY-MM-DD"T"HH24:MI:SS"Z"')::int
        )
        THEN ERROR('this write grant can only be used after dwg_not_before and before dwg_not_after')
```

Reply:
Applied.